PlantPal/PlantPal/PlantPalApp.swift

```diff
@@ -9,9 +9,19 @@ import SwiftUI
 
 @main
 struct PlantPalApp: App {
+    // MARK: - PROPERTIES
+    @StateObject private var viewModel = WelcomeScreenViewModel()
+
     var body: some Scene {
         WindowGroup {
-            ContentView()
+            if viewModel.savedApiKey.isEmpty == false {
+                ScanView()
+                    .environmentObject(viewModel)
```

Comment:
Nie podoba mi się wykorzystanie environmentObject. Tworzy to kilka problemów, o których sobie pogadamy na spotkaniu. Pokażę ci też w jaki sposób bym do tego podszedł.